tracer/src/Datadog.Trace/IAST/Settings/IastSettings.cs

```diff
@@ -16,15 +16,19 @@ internal class IastSettings
     public static readonly int VulnerabilitiesPerRequestDefault = 2;
     public static readonly int MaxConcurrentRequestDefault = 2;
     public static readonly int RequestSamplingDefault = 30;
+    private bool _callsiteEnabled = true;
 
     public IastSettings(IConfigurationSource source)
     {
         WeakCipherAlgorithms = source?.GetString(ConfigurationKeys.Iast.WeakCipherAlgorithms) ?? WeakCipherAlgorithmsDefault;
         WeakCipherAlgorithmsArray = WeakCipherAlgorithms.Split(new char[] { ',' }, System.StringSplitOptions.RemoveEmptyEntries);
         WeakHashAlgorithms = source?.GetString(ConfigurationKeys.Iast.WeakHashAlgorithms) ?? WeakHashAlgorithmsDefault;
         WeakHashAlgorithmsArray = WeakHashAlgorithms.Split(new char[] { ',' }, System.StringSplitOptions.RemoveEmptyEntries);
-        Enabled = (source?.GetBool(ConfigurationKeys.Iast.Enabled) ?? false) &&
-            (WeakHashAlgorithmsArray.Length > 0 || WeakCipherAlgorithmsArray.Length > 0);
+        Enabled = (source?.GetBool(ConfigurationKeys.Iast.Enabled) ?? false);
+        _callsiteEnabled = (source?.GetBool(ConfigurationKeys.Iast.CallsiteEnabled) ?? true);
+#if DEBUG
+        _callsiteEnabled &= (source?.GetBool("DD_CSITE_TESTMODE") ?? false);
```

Comment:
don't understand the why we want a second variable for debug builds?

Reply:
It was for testing aspects in `Private.Corelib`, but I'll rework this.